lib/errors.ts

```diff
@@ -132,6 +132,7 @@ const messages: {
 };
 
 const EXPECTED_ERROR_REGEXES = [
+	/^BalenaAmbiguousApplication:/, // balena-sdk
```

Comment:
This error is raised is someone runs, for example:
```
balena app rm qemux86 -t qemux86-64
```
But not if I run:
```
balena app rm gh_paulo_castro/qemux86
```

Reply:
For these expected errors can they not also be handled via catching the specific error class rather than a regex?

Reply:
I merged the PR just a minute before you added a review. :-)  Re catching the specific error class, it may be possible by using a custom `instanceOf` function, see:
https://github.com/balena-io/balena-cli/blob/v11.35.0/lib/errors.ts#L53

There was a bug fixed recently where the plain `instanceof` operator would fail to match error classes because there were multiple installations (multiple versions) of `balena-errors` in the `npm-shrinkwrap.json` file, such that, for example, `BalenaAmbiguousApplication` may be raised from such a class from a different npm package version than the npm package version providing `BalenaAmbiguousApplication` for use by `instanceof` in `errors.ts`. In other words, as I understand it, the same class may be treated as a different class if they come from different `node_modules` folders...